src/sqlancer/datafusion/DataFusionUtil.java

```diff
@@ -0,0 +1,193 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+import sqlancer.datafusion.DataFusionProvider.DataFusionGlobalState;
+
+public final class DataFusionUtil {
+    private DataFusionUtil() {
+        dfAssert(false, "Utility class cannot be instantiated");
+    }
+
+    // Display tables in `fromTableNames`
+    public static String displayTables(DataFusionGlobalState state, List<String> fromTableNames) {
+        StringBuilder resultStringBuilder = new StringBuilder();
+        for (String tableName : fromTableNames) {
+            String query = String.format("select * from %s", tableName);
+            try (Statement stat = state.getConnection().createStatement();
+                    ResultSet wholeTable = stat.executeQuery(query)) {
+
+                ResultSetMetaData metaData = wholeTable.getMetaData();
+                int columnCount = metaData.getColumnCount();
+
+                resultStringBuilder.append("Table: ").append(tableName).append("\n");
+                for (int i = 1; i <= columnCount; i++) {
+                    resultStringBuilder.append(metaData.getColumnName(i)).append(" (")
+                            .append(metaData.getColumnTypeName(i)).append(")");
+                    if (i < columnCount) {
+                        resultStringBuilder.append(", ");
+                    }
+                }
+                resultStringBuilder.append("\n");
+
+                while (wholeTable.next()) {
+                    for (int i = 1; i <= columnCount; i++) {
+                        resultStringBuilder.append(wholeTable.getString(i));
+                        if (i < columnCount) {
+                            resultStringBuilder.append(", ");
+                        }
+                    }
+                    resultStringBuilder.append("\n");
+                }
+                resultStringBuilder.append("----------------------------------------\n\n");
+
+            } catch (SQLException err) {
+                resultStringBuilder.append("Table: ").append(tableName).append("\n");
+                resultStringBuilder.append("----------------------------------------\n\n");
+                // resultStringBuilder.append("Error retrieving data from table ").append(tableName).append(":
+                // ").append(err.getMessage()).append("\n");
+            }
+        }
+
+        return resultStringBuilder.toString();
+    }
+
+    public static void dfAssert(boolean condition, String message) {
+        if (!condition) {
+            String methodName = Thread.currentThread().getStackTrace()[2].getMethodName();
+            System.err.println("DataFusion assertion failed in function '" + methodName + "': " + message);
+            exit(1);
```

Comment:
Could this be a `new AssertionError()` with the above error message or so? It would be best to avoid direct calls to `exit` (also in other instances).